sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md

```diff
@@ -225,35 +253,57 @@ Now in `Azure.Messaging.ServiceBus`, we simplfify this by giving session variant
+
+    // By default, there will be a single concurrent call per session. I can
+    // increase that here to enable parallel processing within each session.
+    MaxConcurrentCallsPerSession = 2,
+
+    // Processing can be optionally limited to a subset of session Ids.
+    SessionIds = { "my-session", "your-session" },
 };
-ServiceBusSessionProcessor processor = client.CreateSessionProcessor(queueName);
+
+// create a session processor that we can use to process the messages
+ServiceBusSessionProcessor processor = client.CreateSessionProcessor(queueName, options);
 ```
 
 The below code snippet shows you the session variation of the receiver. Please note that creating a session receiver is an async operation because the library will need to get a lock on the session by connecting to the service first.
 
-```cs
-// create a receiver to receive events from the next available session
+Create a receiver that will receive from the next available session:
+```C# Snippet:ServiceBusReceiveNextSession
 ServiceBusSessionReceiver receiver = await client.AcceptNextSessionAsync(queueName);
 
-// create a receiver to receive events from the given session
-ServiceBusSessionReceiver receiver = await client.AcceptSessionAsync(queueName, "my-session");
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveMessageAsync();
+Console.WriteLine(receivedMessage.SessionId);
 ```
 
+Create a receiver that will receive from a specific session:
+```C# Snippet:ServiceBusReceiveFromSpecificSession
+// create a receiver specifying a particular session
+ServiceBusSessionReceiver receiver = await client.AcceptSessionAsync(queueName, "Session2");
+
+// the received message is a different type as it contains some service set properties
+ServiceBusReceivedMessage receivedMessage = await receiver.ReceiveMessageAsync();
+Console.WriteLine(receivedMessage.SessionId);
+```
+
+## Known Gaps from Previous Library
+There are a few features that have yet to be implemented in `Azure.Messaging.ServiceBus`, but were present in the previous library. The plan is to add these features in upcoming releases (unless otherwise noted), but they will not be available at the time of GA:
+- **Cross entity transactions** - In the previous library, transactions could work across multiple entities by leveraging the [`viaEntityPath`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageSender.cs#L118) parameter of the `MessageSender` constructor. The service is planning to make future updates that would make it possible to do cross-entity transactions without specificing a "via" entity. The `viaEntityPath` parameter will still continue to work in the same way in the previous library, but will not be necessary in the new library. Support for this feature can be tracked via https://github.com/Azure/azure-sdk-for-net/issues/17355.
```

Comment:
Any reason to not mention `SendVia` here?
```suggestion
- **Cross entity transactions** - In the previous library, transactions could work across multiple entities by leveraging the constructor parameter `viaEntityPath` for `MessageSender` or the `SendVia` option in `QueueClient`/`TopicClient` constructors. The service is planning to make backward compatible updates that would make it possible to do cross-entity transactions without specificing a "via" entity. The new library plans to take advantage of this new feature and therefore, the cross entity transactions feature will be available in the upcoming release instead of the current version 7.0.0. Support for this feature can be tracked via https://github.com/Azure/azure-sdk-for-net/issues/17355.
```

Reply:
Where are you seeing the `SendVia` option in the QueueClient/TopicClient?

Reply:
My bad... We always referred to it as `SendVia` that I assumed that it was the name of the options passed to one of the client constructors